server/game/cards/characters/01/jonsnow.js

```diff
@@ -6,10 +6,10 @@ class JonSnow extends DrawCard {
     constructor(owner, cardData) {
         super(owner, cardData);
 
-        this.registerEvents(['onAttackersDeclared']);
+        this.registerEvents(['onChallengeInitiated']);
     }
 
-    onAttackersDeclared(event, challenge) {
+    onChallengeInitiated(event, challenge) {
```

Comment:
Since this is now a merged event, `challenge` doesn't exist as a second param, so it should be removed. Need to update usages from `challenge` to `event.challenge` in the method (or simply define `let challenge = event.challenge` as the first line in the method to reduce change noise)